src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -90,19 +98,19 @@ public void enqueue(KafkaMessage message) {
 		}
 	}
 
-	@Override
 	public void start() {
 		this.running = true;
 	}
 
-	@Override
-	public void stop() {
+	public void stop(long stopTimeout) {
+		shutdownLatch = new CountDownLatch(1);
 		this.running = false;
```

Comment:
I don't think we should. Processing the messages can take fairly long and we'd want to return as soon as possible. Given that no offsets get updated as a result, we can replay the.